lib/grp.gi

```diff
@@ -4620,6 +4629,23 @@ InstallMethod( MaximalNormalSubgroups, "for simple groups",
               function(G) return [ TrivialSubgroup(G) ]; end);
 
 
+#############################################################################
+##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "general method selection",
+              [ IsGroup ],
+    function(G)
+
+    if 0 in AbelianInvariants(G) then
+      # (p) is a maximal normal subgroup in Z for every prime p
+      Error("number of maximal normal subgroups is infinity");
```

Comment:
@hungaborhorvath : I often find the easiest thing to do is to run the test without an output, and then cut+paste the output GAP gives you (when it tells you the test is wrong). The output will look something like:

```
gap> 1/0;
Error, Rational operations: <divisor> must not be zero
```

Note how the line / file / backtrace are all missing (they are disabled when you test, specifically for this reason).